oolab/src/test/java/agh/ics/oop/WorldIT.java

```diff
@@ -0,0 +1,42 @@
+package agh.ics.oop;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class WorldIT {
+
+    @Test
+    void dataTest(){
+        OptionsParser p = new OptionsParser();
+        assertArrayEquals(new MoveDirection[]{MoveDirection.FORWARD, MoveDirection.RIGHT, MoveDirection.BACKWARD,
+                MoveDirection.RIGHT, MoveDirection.LEFT},
+                p.parse(new String[]{"f", "r", "v", "b", "r", "l"}));
+    }
+
+    @Test
+    void orientationTest(){
+        Animal a1 = new Animal();
+        OptionsParser parser1 = new OptionsParser();
+        MoveDirection[] dir1 = parser1.parse(new String[]{"l", "r", "r", "u", "r", "l"});
+        for(MoveDirection direction : dir1) a1.move(direction);
+        assertEquals("Orientacja: Wschód, Pozycja: (2,2)", a1.toString());
+    }
+
+    @Test
+    void movementTest(){
+        Animal a2 = new Animal();
+        OptionsParser parser2 = new OptionsParser();
+        MoveDirection[] dir2 = parser2.parse(new String[]{"l", "f", "r", "f", "l", "p"});
+        for(MoveDirection direction : dir2) a2.move(direction);
+        assertEquals("Orientacja: Zachód, Pozycja: (1,3)", a2.toString());
```

Comment:
ok, przy czym w tego typu teście sprawdzamy rezultat końcowy całej sekwencji - jeśli po drodze program popełniłby kilka sprzecznych ze sobą błędów i wyszłoby na końcu na plus to test dalej by przechodził.